src/lua/init.lua

```diff
@@ -194,15 +227,19 @@ local function search(name)
 end
 
 -- loader_preload 1
-table.insert(package.loaders, 2, cwd_loader_func(false))
-table.insert(package.loaders, 3, cwd_loader_func(true))
-table.insert(package.loaders, 4, rocks_loader_func(false))
-table.insert(package.loaders, 5, rocks_loader_func(true))
--- package.path   6
--- package.cpath  7
--- croot          8
+table.insert(package.loaders, 2, get_loader_func({path_fn = fio.cwd, is_lib = false}))
+table.insert(package.loaders, 3, get_loader_func({path_fn = fio.cwd, is_lib = true}))
+table.insert(package.loaders, 4, get_loader_func({path_fn = approot, is_lib = false}))
+table.insert(package.loaders, 5, get_loader_func({path_fn = approot, is_lib = true}))
+table.insert(package.loaders, 6, get_loader_func({path_fn = get_rocksluadir, is_lib = false, traverse = true}))
+table.insert(package.loaders, 7, get_loader_func({path_fn = get_rockslibdir, is_lib = true, traverse = true}))
```

Comment:
This bunch looks strange.

1. `is_lib` option partially overlaps what is passed to `path_fn`.
2. why does it accept function as an argument?
3. when approot is not set loaders are duplicated

Reply:
> 1. is_lib option partially overlaps what is passed to path_fn.

Sorry, I didn't get the part about overlapping. `is_lib` flag lets loader function to define both searchpath templates (i.e. `?.so` for libs) and the way how this module will be loaded (i.e. using `package.loadlib` for libs). `path_fn` is a simple function, which returns some path which will be later used to append lib or lua template.

> 2. why does it accept function as an argument?

function for paths are needed to be able to get an actual path at runtime. I.e. current working directory may be changed during runtime, so we need to get the right one when doing `require`

> 3. when approot is not set loaders are duplicated

You are right. But I don't know how to improve it right now. We need to add this loader anyway.

Reply:
Refactored `get_loader_func` api to eliminate `is_lib` parameter. Now it looks much more readable